src/components/BarChart/BarChart.tsx

```diff
@@ -0,0 +1,238 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import * as d3 from 'd3';
+import { IBarChartProps, IBarChartData } from './BarChart.props';
+import './BarChart.scss';
+
+enum Update {
+    Data = 1,
+    Dimension
+};
+
+export class BarChart extends React.Component<IBarChartProps, null> {
+
+    public static defaultProps = {
+        id: '',
+        barColor: '#889ac4',
+        hovColor: '#b8c7e8',
+        data: [],
+        tipText: (data: IBarChartData) => ('Frequency of ' + data.argument + ' is : ' + data.frequency),
+        xAxisFormat: () => null,
+        width: 500,
+        height: 300,
+        minWidth: 300,
+        maxWidth: 1000,
+        isResponsive: false,
+        onClick: () => { return; }
+    };
+
+    public refs: {
+        [argument: string]: (Element);
+        container: (HTMLInputElement);
+    };
+
+    private margin = { top: 50, right: 50, bottom: 50, left: 50 };
+    private width = this.props.width - this.margin.right - this.margin.left;
+    private height = this.props.height - this.margin.top - this.margin.bottom;
+    private x;
+    private y;
+    private focus;
+    private svg;
+    private container;
+
+    constructor(props: IBarChartProps) {
+        super(props);
+
+        this.x = this.generateX();
+        this.y = this.generateY();
+
+        // if (this.props.isResponsive) {
+        //     this.width = window.innerWidth / 2;
+        //     this.height = 0.7 * this.width;
+        // }
+    }
+
+    public componentWillReceiveProps(nextProps: IBarChartProps) {
+        // console.log('next props', nextProps);
+
+        // see what props have changed and set appropriate enum based on that
+    }
+
+    public shouldComponentUpdate(nextProps: IBarChartProps, nextState: null) {
+        if (this.props.id !== nextProps.id ||
+            this.props.width !== nextProps.width ||
+            this.props.height !== nextProps.height ||
+            this.props.barColor !== nextProps.barColor) {
+                return true;
+            }
+        if (!this.arraysEqual(this.props.data, nextProps.data)) { return true; }

```

Comment:
are you sure you are passing down the props that shallow compare know how to compare?
Deep comparing the arrays in onShouldComponentUpdate servers no purpose because it is very costly operation running frequently